### PR TITLE
CBL-1166: test: add listener test with import identity

### PR DESCRIPTION
**Pull request by jayahariv**
* reused the store and update cert methods to separate helper class
* add test which does imported the certs and prvate key, and verify replicating a doc

### CODE REVIEW

File: Swift/Tests/ReplicatorTest+CustomConflict.swift [hunk 1]
```diff
@@ -429,7 +429,8 @@ class ReplicatorTest_CustomConflict: ReplicatorTest {
         XCTAssert(db.document(withID: docID)!.toDictionary() == remoteData)
     }
     
-    func testConflictResolutionDefault() throws {
+    // TODO: https://issues.couchbase.com/browse/CBL-1174
+    func _testConflictResolutionDefault() throws {
```
Comment (jayahariv) [inline Swift/Tests/ReplicatorTest+CustomConflict.swift:433]: this is for making PR validation success

File: Swift/Tests/ReplicatorTest+CustomConflict.swift [hunk 2]
```diff
@@ -429,7 +429,8 @@ class ReplicatorTest_CustomConflict: ReplicatorTest {
         XCTAssert(db.document(withID: docID)!.toDictionary() == remoteData)
     }
     
-    func testConflictResolutionDefault() throws {
+    // TODO: https://issues.couchbase.com/browse/CBL-1174
```
Comment (jayahariv) [inline Swift/Tests/ReplicatorTest+CustomConflict.swift:432]:
```suggestion
```

File: Swift/Tests/ReplicatorTest+CustomConflict.swift (hunk 1 above)
Comment (jayahariv) [inline Swift/Tests/ReplicatorTest+CustomConflict.swift:433]:
```suggestion
    func testConflictResolutionDefault() throws {
```

**Comment by jayahariv** [comment]
ignore the PR validation failure, since its not rebase to master.

File: Objective-C/Tests/URLEndpointListenerTest.m [hunk 3]
```diff
@@ -1153,20 +1170,77 @@ - (void) testAcceptOnlySelfSignedCertificateWithPinnedCertificate {
...
-        
-        NSString* cn1 = (NSString*)CFBridgingRelease(cnRef1);
-        NSString* cn2 = (NSString*)CFBridgingRelease(cnRef2);
-        AssertEqualObjects(cn1, cn2);
-    }
-}
-
-- (void) releaseCF: (CFTypeRef)ref {
-    if (ref != NULL) CFRelease(ref);
+- (void) testListenerWithImportIdentity {
```
Comment (pasin) [inline Objective-C/Tests/URLEndpointListenerTest.m:1173]: This is the test that tests create identity with identity. Can you also add another test that uses TLSIdentity's importIdentity?

**Comment by jayahariv** [comment]
Closed and opened a fresh PR, minim changes.

**Pull request closed without merging**